os/arch/arm/src/armv7-m/up_svcall.c

```diff
@@ -350,7 +350,22 @@ int up_svcall(int irq, FAR void *context, FAR void *arg)
 		 * unprivileged mode.
 		 */
 
-		regs[REG_PC] = (uint32_t)USERSPACE->task_startup;
+#ifdef CONFIG_APP_BINARY_SEPARATION
+		/* While starting loadable apps, we cannot go through the
+		* USERSPACE->task_startup method. Instead we pick the PC value
+		* from the app's userspace object stored in its tcb.
+		*
+		* Here, we check if this task is a loadable app (non-zero ram_start)
+		*/
+		if (((struct tcb_s*)sched_self())->ram_start) {
```

Comment:
Please add space between `tcb_s` and `*`.

Reply:
fixed